stake-o-matic/src/main.rs

```diff
@@ -653,6 +695,35 @@ fn main() -> Result<(), Box<dyn error::Error>> {
         process::exit(1);
     });
 
+    let cluster_nodes_with_old_version: HashSet<String> = match config.minimum_release_version {
+        Some(ref minimum_release_version) => rpc_client
+            .get_cluster_nodes()?
+            .into_iter()
+            .filter_map(|rpc_contact_info| {
+                if let Ok(pubkey) = Pubkey::from_str(&rpc_contact_info.pubkey) {
+                    if config.validator_list.contains(&pubkey) {
+                        if let Some(ref version) = rpc_contact_info.version {
+                            if let Ok(semver) = semver::Version::parse(version) {
+                                if semver < *minimum_release_version {
```

Comment:
This might be more readable as a `filter/filter_map` chain

Reply:
I had that first actually and was so proud at how functional I was being.  But I found it harder to read

Reply:
Interesting!  I'll allow it :slightly_smiling_face: